fix uses of map() for py3 compatibility

anthrotype opened the pull request:
In pythoh 3, `map()` returns an iterator object instead of a list.
After this patch, I can run compreffor with python 3.

adrientetar commented on compreffor/cxxCompressor.py:
```diff
@@ -259,7 +259,7 @@ def compreff(font, verbose=False, use_lib=False, **kwargs):
     # in order of charset
-    chstrings = map(lambda x: x.program, td.CharStrings.values())
+    chstrings = [x.program for x in td.CharStrings.values()]
     map(lambda x: Compreffor.collapse_hintmask(x), chstrings)
```
Does this run the generator?

anthrotype replied:
no, it didn't run. I just replaced it with a for-loop with https://github.com/googlei18n/compreffor/pull/13/commits/daea41f747f133dbd80ff7f4599fe540dcf0975a

anthrotype commented:
@adrientetar @behdad the issue with compreffor producing different binary CFF tables in python 2 vs python 3 is now fixed. The bug was in `fontTools.cffLib.IndexedStrings.getSID`, and it's now fixed:
https://github.com/behdad/fonttools/commit/6c1c2a44d1dcfb0b8001edaf3ddc442cec2e0a11

I think we can now merge this PR branch.
